From 419a869700fa6159b6eae19f3b3bb83dfdf707bd Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 20 May 2020 11:40:28 +0300
Subject: [PATCH] [SW-9973] spi: move status register bits to spi.h

Move the SR bits to the header file, so device specific code can build a
device specific protection mask from them.

Change-Id: I0d4a72caa22cefd6014a6665ed34ff1210e08aad
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/spi.c b/common/spi.c
index e057c87..d35f70d 100644
--- a/common/spi.c
+++ b/common/spi.c
@@ -31,21 +31,6 @@
 #define CMD_WRITE_DISABLE		0x04
 #define CMD_READ_STATUS			0x05
 #define CMD_WRITE_ENABLE		0x06
-#define READ_STATUS_WIP			BIT(0)
-#define READ_STATUS_BP0			BIT(2)
-#define READ_STATUS_BP1			BIT(3)
-#define READ_STATUS_BP2			BIT(4)
-#define READ_STATUS_BP3			BIT(5)
-#define READ_STATUS_SRWD		BIT(7)
-#define READ_STATUS_BP0_3		(READ_STATUS_BP0 | READ_STATUS_BP1 | \
-					READ_STATUS_BP2 | READ_STATUS_BP3)
-#define ALL_PROT_MASK			(READ_STATUS_BP0_3 | READ_STATUS_SRWD)
-
-/* The PROT_MASK is meant to be overridden by device specific setting */
-#ifndef PROT_MASK
-#define PROT_MASK			ALL_PROT_MASK
-#endif
-
 #define frame_size_bits(b)		((b) << 16)
 
 static unsigned int spi_regs;
diff --git a/common/spi.h b/common/spi.h
index ebbbac6..faa710d 100644
--- a/common/spi.h
+++ b/common/spi.h
@@ -8,6 +8,22 @@
 #ifndef _HL_SPI_H_
 #define _HL_SPI_H_
 
+/* The Status Register (SR) in use bits definition */
+#define READ_STATUS_WIP			BIT(0)
+#define READ_STATUS_BP0			BIT(2)
+#define READ_STATUS_BP1			BIT(3)
+#define READ_STATUS_BP2			BIT(4)
+#define READ_STATUS_BP3			BIT(5)
+#define READ_STATUS_SRWD		BIT(7)
+#define READ_STATUS_BP0_3		(READ_STATUS_BP0 | READ_STATUS_BP1 | \
+					READ_STATUS_BP2 | READ_STATUS_BP3)
+#define ALL_PROT_MASK			(READ_STATUS_BP0_3 | READ_STATUS_SRWD)
+
+/* The PROT_MASK is meant to be overridden by device specific setting */
+#ifndef PROT_MASK
+#define PROT_MASK			ALL_PROT_MASK
+#endif
+
 void spi_copy(void *dst, unsigned int offset, unsigned int size, char *name);
 u32 get_cs_pad_sel(void);
 
